src/libmongoc/src/mongoc/mongoc-client-side-encryption.c

```diff
@@ -88,6 +88,8 @@ void
 mongoc_auto_encryption_opts_set_keyvault_client (
    mongoc_auto_encryption_opts_t *opts, mongoc_client_t *client)
 {
+   BSON_ASSERT_PARAM (client);
```

Comment:
```suggestion
```

Suggest removing this assertion. A caller may be relying on passing `NULL` to clear the option.